src/backend/PortabilityService.DependencyInjection/InjectBinding.cs

```diff
@@ -0,0 +1,71 @@
+﻿// MIT License
+//
+// Copyright(c) 2017 Boris Wilhelms
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading.Tasks;
+
+namespace DependencyInjection
+{
+
+    public class InjectBinding : IBinding
+    {
+        private readonly Type _type;
+        private readonly IServiceProvider _serviceProvider;
+
+        public InjectBinding(IServiceProvider serviceProvider, Type type)
+        {
+            _type = type;
+            _serviceProvider = serviceProvider;
+        }
+
+        public bool FromAttribute => true;
+
+        public Task<IValueProvider> BindAsync(object value, ValueBindingContext context) =>
+            Task.FromResult((IValueProvider)new InjectValueProvider(value));
+
+        public async Task<IValueProvider> BindAsync(BindingContext context)
+        {
+            await Task.Yield();
```

Comment:
Curious why there is a Task.Yield() here?

Reply:
The author mentioned that it isn't really needed. But I'd like to keep his code intact if possible.

---
In reply to: [178093177](https://github.com/Microsoft/dotnet-apiport/pull/598#discussion_r178093177) [](ancestors = 178093177)

Reply:
Fixed

---
In reply to: [179242971](https://github.com/Microsoft/dotnet-apiport/pull/598#discussion_r179242971) [](ancestors = 179242971,178093177)